public/Get-VagrantEnvironment.ps1

```diff
@@ -16,30 +16,47 @@ function Get-VagrantEnvironment {
     #>
     
     [cmdletBinding()]
-    Param()
+    Param(
+        [switch]
+        $All
+    )
 
     process {
+        if ($All) {
+            $environments = vagrant global-status
         
-        $environments = vagrant global-status
-        
-        $environments = $environments[2..$($environments.Length-8)]
-
-        Foreach($environment in $environments){
+            $environments = $environments[2..$($environments.Length - 8)]
+            Foreach ($environment in $environments) {
 
-            if($environment -match "^(?<id>[\w]+)\s+(?<name>[\w\-._]+)\s+(?<provider>[\w]+)\s+(?<state>[\w]+)\s+(?<path>[\/\-\w:]+)"){
-           
-                [pscustomobject]@{
-                    Id = $matches.id
-                    Name = $matches.name
-                    Provider = $matches.provider
-                    State = $matches.state
-                    Path = $matches.path
-                }
+                if ($environment -match "^(?<id>[\w]+)\s+(?<name>[\w\-._]+)\s+(?<provider>[\w]+)\s+(?<state>[\w]+)\s+(?<path>[\/\-\w:]+)") {
+                    [pscustomobject]@{
+                        Id       = $matches.id
+                        Name     = $matches.name
+                        Provider = $matches.provider
+                        State    = $matches.state
+                        Path     = $matches.path
+                    }
             
+                }
+                
             }
+        }
+        else {
+            $environments = vagrant status
             
+            $environments = $environments[2..$($environments.Length - 4)]

```

Comment:
This is basically not needed (and will be incorrect depending on the state of the vagrant environment). We're matching the regex below, all of the mentioned lines won't match the regex...